lib/colorado_lottery.rb

```diff
@@ -0,0 +1,45 @@
+class ColoradoLottery
+  attr_accessor :registered_contestants, :winners, :current_contestants
+  def initialize
+    @registered_contestants = {}
+    @winners = []
+    @current_contestants = {}
+  end
+
+  def interested_and_18?(contestant, game)
+    contestant.game_interests.include?(game.name) && (contestant.age >= 18)
+  end
+
+  def can_register?(contestant, game)
+
+    interested_and_18?(contestant, game) &&
+    ((game.national_drawing?) || (contestant.state_of_residence == "CO"))
+  end
+
+  def register_contestant(contestant, game)
+
+     if registered_contestants[game.name] == nil #&& (contestant.spending_money > game.cost)
+      registered_contestants[game.name] = []
+      registered_contestants[game.name] << contestant
+      #contestant.charge_contestant(game.cost)
+    elsif #registered_contestants[game.name] != nil #&& (contestant.spending_money > game.cost)
+      registered_contestants[game.name] << contestant
+      #contestant.charge_contestant(game.cost)
+    end
+  end
+
+  def eligible_contestants(game)
+    contestants = registered_contestants[game.name]
+    contestants
```

Comment:
not quite-- great job getting the contestants for a specific game using the key.  however, you needed to check each of these contestants to see which of them had enough spending money for the game. ("#eligible_contestants is a list of all the contestants who have been registered to play a given game and that have more spending_money than the cost.")

```
  def eligible_contestants(game)
    registered_contestants[game.name].find_all {|contestant| contestant.spending_money >= game.cost}
  end
```